src/server.c

```diff
@@ -1219,6 +1222,8 @@ static void setup_ssl_sasl_and_open(qd_connection_t *ctx)
                 qd_log(ct->server->log_source, QD_LOG_ERROR,
                        "SSL local configuration failed for %s:%s",
                        config->host, config->port);
+                pn_ssl_domain_free(domain);
+                return;
```

Comment:
could we populate a boolean inside these if statements and check it finally and then return ? That would keep the code behaving as it was earlier.

Reply:
yes. fixed.